pkg/resource/function/hooks.go

```diff
@@ -179,6 +179,38 @@ func (rm *resourceManager) updateFunctionConfiguration(
 		}
 	}
 
+	if delta.DifferentAt("Spec.Environment") {
+		environment := &svcsdk.Environment{}
+		if dspec.Environment != nil {
+			environment.Variables = dspec.Environment.DeepCopy().Variables
+		}
+		input.Environment = environment
+	}
+
+	if delta.DifferentAt("Spec.EphemeralStorage") {
+		ephemeralStorage := &svcsdk.EphemeralStorage{}
+		if dspec.EphemeralStorage != nil {
+			ephemeralStorageCopy := dspec.EphemeralStorage.DeepCopy()
+			ephemeralStorage.Size = ephemeralStorageCopy.Size
+		}
+		input.EphemeralStorage = ephemeralStorage
+	}
+
+	if delta.DifferentAt("Spec.FileSystemConfigs") {
+		fileSystemConfigs := []*svcsdk.FileSystemConfig{}
+		if len(dspec.FileSystemConfigs) > 0 {
+			for _, elem := range dspec.FileSystemConfigs {
+				elemCopy := elem.DeepCopy()
+				fscElem := &svcsdk.FileSystemConfig{
+					Arn:            elemCopy.ARN,
+					LocalMountPath: elemCopy.LocalMountPath,
+				}
+				fileSystemConfigs = append(fileSystemConfigs, fscElem)
+			}
+			input.FileSystemConfigs = fileSystemConfigs
+		}
+	}
```

Comment:
Thanks @Vandita2020 ! How are environement/ephemeralStorage/FileSystemConfigs/Timeouts affecting to layers?

Reply:
Hey, I have just rearranged them according to the order in the API. Please ignore them, no changes for them.